using pparams consistently with byron spec

JaredCorduan opened the pull request:
The PR aims to get consistency between the two Byron Specs and the Shelley spec in how protocol parameters are handled. It's pretty minimal.

* Instead of defining `PParams` as a new abstract type, we now reference the Byron ledger spec, so that the type is `PPm |-> Value`.
* The section on protocol parameters now includes all the Byron parameters. I renamed two of the Shelley parameters so that they would match Byron: `maxBHSize` is now `maxHeaderSize`, and `maxBBSize` is now `maxBlockSize`.
* The Shelley spec now computes the min fee like Byron, in terms of the two parameters `a` and `b`.
* The protocol parameters table got too big, so I split off the helper functions off into their own table.

One thing to note is that I did not change the style of using the protocol parameters. In the Byron ledger spec, two styles are used: accessor style like `foo pps` is used in Figure 3 of the Byron ledger spec, and map predicates like `foo |-> bar in pps` are used elsewhere. I like the flexibility of being able to use either! The Shelley spec is written in the first style, and so I kept it this way.

closes #443
closes #102

mgudemann commented on shelley/chain-and-ledger/formal-spec/protocol-parameters.tex:
```diff
@@ -1,24 +1,27 @@
+Two global constants are defined, $\SlotsPerEpoch$ and $\SlotsPerKESPeriod$,
+representing the number of slots in an epoch/KES period.  Cycles will be used in
+the key evolving signatures.  As global constants, these values can only be
+changed by updating the software.
```
maybe remove the two double spaces in this paragraph

mgudemann commented on shelley/chain-and-ledger/formal-spec/protocol-parameters.tex:
```diff
@@ -114,9 +123,19 @@ \section{Protocol Parameters}
   %
   \begin{align*}
+    \fun{minfee} & \in \PParams \to \Tx \to \mathbb{Z} & \text{minimum fee}\\
+    \fun{minfee} & ~\var{pps}~\var{tx} =
```
```minfee``` is defined above as an abstract function with a different type, evaluating to `Coin`

JaredCorduan replied:
good catch, I removed the abstract one

JaredCorduan commented:
@dnadales I updated the notation section in Byron to make clear the partial function notation. And then copied that description for Shelley. I also renamed "fibre" in Byron to "inverse image" (that's a change I made a while back for Shelley).
